python/meep.i

```diff
@@ -366,12 +366,20 @@ PyObject *_get_dft_array(meep::fields *f, dft_type dft, meep::component c, int n
     int dims[3];
     std::complex<double> *dft_arr = f->get_dft_array(dft, c, num_freq, &rank, dims);
 
+    size_t length = 1;
     npy_intp *arr_dims = new npy_intp[rank];
     for (int i = 0; i < rank; ++i) {
         arr_dims[i] = dims[i];
+        length *= dims[i];
     }
 
-    PyObject *py_arr = PyArray_SimpleNewFromData(rank, arr_dims, NPY_CDOUBLE, dft_arr);
+    // Allocate our own version of dft_arr with malloc because
+    // numpy will eventually call free on it.
```

Comment:
Seems like it would be simpler to just do:
```c++
PyObject *py_arr = PyArray_SimpleNew(rank, arr_dims, NPY_CDOUBLE);
memcpy(PyArray_DATA(py_arr), dft_arr, sizeof(std::complex<double>) * length);
```
i.e. to let Python do the allocation.

Reply:
Fixed.